src/plans/D3MCompoundPlan.sol

```diff
@@ -37,163 +37,162 @@
-        return _calculateTargetSupply(targetInterestRate, cDai.totalBorrows());
-    }
-
     function _calculateTargetSupply(uint256 targetInterestRate, uint256 borrows) internal view returns (uint256) {
-        uint256 kink                   = rateModel.kink();
-        uint256 multiplierPerBlock     = rateModel.multiplierPerBlock();
-        uint256 baseRatePerBlock       = rateModel.baseRatePerBlock();
-        uint256 jumpMultiplierPerBlock = rateModel.jumpMultiplierPerBlock();
+        uint256 kink                   = tack.kink();
+        uint256 multiplierPerBlock     = tack.multiplierPerBlock();
+        uint256 baseRatePerBlock       = tack.baseRatePerBlock();
+        uint256 jumpMultiplierPerBlock = tack.jumpMultiplierPerBlock();
 
-        uint256 normalRate = _add(_wmul(kink, multiplierPerBlock), baseRatePerBlock);
+        // The normal rate is a Compound term for the rate at kink utillization
+        uint256 normalRate = _wmul(kink, multiplierPerBlock) + baseRatePerBlock;
 
         uint256 targetUtil;
         if (targetInterestRate > normalRate) {
             if (jumpMultiplierPerBlock == 0) return 0; // illegal rate, max is normal rate for this case
-            targetUtil = _add(kink, _wdiv(targetInterestRate - normalRate, jumpMultiplierPerBlock));             // (1)
+            targetUtil = kink + _wdiv(targetInterestRate - normalRate, jumpMultiplierPerBlock); // (1)
         } else if (targetInterestRate > baseRatePerBlock) {
-            targetUtil = _wdiv(targetInterestRate - baseRatePerBlock, multiplierPerBlock);                       // (2)
+            // multiplierPerBlock != 0, as otherwise normalRate == baseRatePerBlock, matching the first case
+            targetUtil = _wdiv(targetInterestRate - baseRatePerBlock, multiplierPerBlock);      // (2)
         } else {
             // if (target == base) => (borrows == 0) => supply does not matter
             // if (target  < base) => illegal rate
             return 0;
         }
 
-        if (targetUtil > WAD) return 0; // illegal rate (unacheivable utilization)
+        return _wdiv(borrows, targetUtil);                                                      // (3)
+    }
+
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
+        uint256 targetInterestRate = barb;
+        if (targetInterestRate == 0) return 0; // De-activated
+
+        uint256 borrows = cDai.totalBorrows();
+        uint256 targetTotalPoolSize = _calculateTargetSupply(targetInterestRate, borrows);
+        uint256 totalPoolSize = cDai.getCash() + borrows - cDai.totalReserves();
 
-        return _wdiv(borrows, targetUtil);                                                                       // (3)
+        if (targetTotalPoolSize >= totalPoolSize) {
+            // Increase debt (or same)
+            return currentAssets + targetTotalPoolSize - totalPoolSize;
```

Comment:
For additional protection against overflow it's better to subtract the delta first before performing the addition.

Reply:
Done in `75d0a7a4f1`